program/src/processor.rs

```diff
@@ -6832,6 +7213,96 @@ impl Processor {
                 msg!("Mango: CancelAllSpotOrders");
                 Self::cancel_all_spot_orders(program_id, accounts, limit)
             }
+            MangoInstruction::EditSpotOrder { cancel_order, cancel_order_size, new_order } => {
+                msg!("Mango: EditSpotOrder");
+                Self::edit_spot_order(
+                    program_id,
+                    accounts,
+                    cancel_order,
+                    cancel_order_size,
+                    new_order,
+                )
+            }
+            MangoInstruction::EditPerpOrderByClientId {
+                price,
+                max_base_quantity,
+                max_quote_quantity,
+                expiry_timestamp,
+                client_order_id,
+                cancel_order_size,
+                side,
+                order_type,
+                reduce_only,
+                limit,
+                invalid_id_ok,
+            } => {
+                msg!("Mango: EditPerpOrderByClientId client_order_id={}", client_order_id);
+                let result = Self::edit_perp_order_by_client_id(
+                    program_id,
+                    accounts,
+                    price,
+                    max_base_quantity,
+                    max_quote_quantity,
+                    expiry_timestamp,
+                    client_order_id,
+                    cancel_order_size,
+                    side,
+                    order_type,
+                    reduce_only,
+                    limit,
+                );
+                if invalid_id_ok {
```

Comment:
hmm. if the previous order is missing, should we assume it was fully filled and adjust the new order accordingly?

Current behavior seems to be:
- if the old order is partially filled, cancel it; adjust new order and place it
- if the old order is missing:
  - if invalid_id_ok is false: error
  - if invalid_id_ok is true: do nothing (i.e. don't place the new order)

To me it sounds like "if the old order is gone (fully filled), then adjust the new order for that and place it" would also make sense.

Reply:
Initially I copied the invalid_id_ok flag to mimic the cancel_perp_order instructions, but now I think it doesn't make much sense for editing orders. To be more straightforward with the user, edit_order instructions should fail if there is no order to edit. The updated behavior is:

- if the old order is partially filled, cancel it; adjust new order and place it
- if the old order is missing (fully filled), instruction will fail because ```book.get_order()``` will fail to find an order and will throw a ```MangoErrorCode::InvalidOrderId```